content/blog/2018-11-11-colocr.md

```diff
@@ -0,0 +1,106 @@
+topicid:
+tags:
+- r
+- package
+- onboarding
+- imaging
+- reproducibility
+---
+
+A few months ago, I wasn't sure what to expect when reading microscopy images. I looked at the accompanying graph to understand the data or the point the authors is trying to make. Often, the graph represents one or more measures of the so-called co-localization, but I couldn’t figure out how to interpret them. It turned out; reading the images is simple. Cells are simultaneously stained by two dyes (say, red and green) for two different proteins. The color turns yellow in the merged image when the two proteins localize in proximity. 
+
+A recent publication from our laboratory included microscopy images and quantification graph. The standard analysis protocol is to use [ImageJ](https://imagej.nih.gov/ij/index.html) with a specialized plug-in for the co-localization analysis such as [coloc2](https://imagej.net/Coloc_2) or [Fiji](https://fiji.sc). The images were processed one at a time; subtracting the background, selecting regions of interest and calculating the co-localization statistics. Luckily for me, I didn’t do the work myself, but it took a good chunk of my colleague precious time. Even worse, she had to repeat this whole process more than once with minor changes in the parameters each time. At the end, there was no way to make this process reproducible, and we included only the final output in an otherwise fully reproducible article.
+
+For those two specific reasons; the need for manual work and the lack of reproducibility, I thought it would be great if I could do this analysis in [R](https://cran.r-project.org). There are several [R](https://cran.r-project.org) packages for image processing such as `EBImage`, `imager` and `magick`. These packages provided the functionality needed for conducting this analysis. So I put the pieces together in an [R](https://cran.r-project.org) package to simplify this analysis for my future self and for others who might find it useful. This is `colocr`, a simple [R](https://cran.r-project.org) package for conducting co-localization analysis of fluorescence microscopy images. In particular, it semi-automates the selections of regions of interest-arguably the most time-consuming step- and comes with a shiny app for visually comparing the outputs of different parameters before writing a final script. 
+
+## How would've we done it differently?
+
+In this recent article I referred to earlier, we assessed the co-localization of several proteins with RKIP (Raf-kinas inhibitor protein) in DU145 prostate cancer cell line. For each of these proteins, we generated multiple frames, a hundred in total. We then processed the images one by one and repeated the process with each minor change in the parameters. Now we can use `colocr`, in just a few lines of code, to apply the same analysis.
+
+A selection of the images were presented in the article in a similar figure.
+
+![](/img/blog-images/2018-11-11-colocr/images.png)
```

Comment:
should there be a heading above the red dye column?

Reply:
The labels for the red dye images are displayed on the left.